### PR TITLE
Use secrets manager to read serverless credentials

### PR DESCRIPTION
**Pull request by alcaeus**
This fixes the serverless build after the credentials were rotated.

Note: the change to use subprocess.exec instead of shell.exec is not related to the original issue, but it's recommended by the evergreen team.

### CODE REVIEW

File: .evergreen/config/functions.yml [hunk 1]
```diff
@@ -171,14 +171,12 @@ functions:
...
       params:
         working_dir: "src"
-        script: |
-          ${PREPARE_SHELL}
-
-          bash ${DRIVERS_TOOLS}/.evergreen/serverless/setup-secrets.sh
-          bash ${DRIVERS_TOOLS}/.evergreen/serverless/create-instance.sh
+        binary: bash
+        args:
+          - ${DRIVERS_TOOLS}/.evergreen/serverless/create-instance.sh
```
Comment (alcaeus) [inline .evergreen/config/functions.yml:179]: setup-secrets is now automatically called in create-instance

File: .evergreen/config/functions.yml [hunk 2]
```diff
@@ -274,12 +274,15 @@ functions:
           export KMS_TLS_CA_FILE="${client_side_encryption_kms_tls_ca_file}"
           export KMS_TLS_CERTIFICATE_KEY_FILE="${client_side_encryption_kms_tls_certificate_key_file}"
           export MONGODB_IS_SERVERLESS=on
-          export MONGODB_USERNAME=${SERVERLESS_ATLAS_USER}
-          export MONGODB_PASSWORD=${SERVERLESS_ATLAS_PASSWORD}
+          export MONGODB_URI="${SERVERLESS_URI}"
           export PATH="${PHP_PATH}/bin:$PATH"
 
+          . ${DRIVERS_TOOLS}/.evergreen/serverless/secrets-export.sh
```
Comment (alcaeus) [inline .evergreen/config/functions.yml:280]: This file is created by setup-secrets and contains the secrets needed to run tests

File: .evergreen/config/functions.yml [hunk 3]
```diff
@@ -274,12 +274,15 @@ functions:
...
+
+          export MONGODB_USERNAME=$SERVERLESS_ATLAS_USER
+          export MONGODB_PASSWORD=$SERVERLESS_ATLAS_PASSWORD
```
Comment (alcaeus) [inline .evergreen/config/functions.yml:283]: FYI, it took me too long to remember that `$FOO` refers to an environment variable named `FOO`, whereas `${FOO}` refers to an evergreen expansion named `FOO` 🤦‍♂️

File: .evergreen/config/functions.yml [hunk 4]
```diff
@@ -274,12 +274,15 @@ functions:
           export KMS_TLS_CA_FILE="${client_side_encryption_kms_tls_ca_file}"
           export KMS_TLS_CERTIFICATE_KEY_FILE="${client_side_encryption_kms_tls_certificate_key_file}"
           export MONGODB_IS_SERVERLESS=on
-          export MONGODB_USERNAME=${SERVERLESS_ATLAS_USER}
-          export MONGODB_PASSWORD=${SERVERLESS_ATLAS_PASSWORD}
+          export MONGODB_URI="${SERVERLESS_URI}"
```
Comment (jmikola) [inline .evergreen/config/functions.yml:277]: Why was this moved to an `export` instead of a direct env var for calling `run-tests.sh`?

**Comment by jmikola** [comment]
Just reconfigured the patch build to include both serverless tasks: https://spruce.mongodb.com/version/65de3efcc9ec44736b863734/tasks

**Comment by alcaeus** [comment]
Rebased on v1.17 and re-scheduled the two serverless tasks in the PR patch.

**Review by jmikola — APPROVED** [review]
My question in https://github.com/mongodb/mongo-php-library/pull/1237#discussion_r1506219613 is outstanding but I won't argue with the build status.

**Comment by alcaeus** [comment]
evergreen keep-definitions
